*** /usr/share/vala-0.10/vapi/mysql.vapi	2010-10-01 11:17:31.000000000 +0200
--- mysql2.vapi	2010-10-25 01:54:59.000000000 +0200
***************
*** 78,97 ****
  		public bool eof ();
  		public Field fetch_field ();
  		public Field fetch_field_direct (uint field_nbr);
! 		public Field[] fetch_fields ();
  		public ulong[] fetch_lengths ();
  		[CCode (array_length = false, array_null_terminated = true)]
  		public unowned string[]? fetch_row ();		
  		public uint fetch_count ();
  		public uint num_fields ();
  		public uint num_rows ();
- 		
  		public bool data_seek (ulong offset);
  	}
  	[CCode (cname = "MYSQL_FIELD")]
  	public struct Field {
  	}
- 	
  	public unowned string get_client_info ();
  	public ulong get_client_version ();
  	public void debug (string msg);
--- 78,137 ----
  		public bool eof ();
  		public Field fetch_field ();
  		public Field fetch_field_direct (uint field_nbr);
! 		[CCode (array_length = false, array_null_terminated = true)]
! 		public unowned Field[]? fetch_fields ();
  		public ulong[] fetch_lengths ();
  		[CCode (array_length = false, array_null_terminated = true)]
  		public unowned string[]? fetch_row ();		
  		public uint fetch_count ();
  		public uint num_fields ();
  		public uint num_rows ();
  		public bool data_seek (ulong offset);
  	}
+ 	[CCode (cname = "enum_field_types", cprefix = "MYSQL_TYPE_", cheader_filename = "mysql/mysql_com.h")]
+ 	public enum FieldType {
+ 		DECIMAL,
+ 		TINY,
+ 		SHORT,
+ 		LONG,
+ 		FLOAT,
+ 		DOUBLE,
+ 		NULL,
+ 		TIMESTAMP,
+ 		LONGLONG,
+ 		INT24,
+ 		DATE,
+ 		TIME,
+ 		DATETIME,
+ 		YEAR,
+ 		NEWDATE,
+ 		VARCHAR,
+ 		BIT,
+ 		NEWDECIMAL=246,
+ 		ENUM=247,
+ 		SET=248,
+ 		TINY_BLOB=249,
+ 		MEDIUM_BLOB=250,
+ 		LONG_BLOB=251,
+ 		BLOB=252,
+ 		VAR_STRING=253,
+ 		STRING=254,
+ 		GEOMETRY=255
+ 	}
  	[CCode (cname = "MYSQL_FIELD")]
  	public struct Field {
+ 		public unowned string name;
+ 		public unowned string org_name;
+ 		public unowned string table;
+ 		public unowned string org_table;
+ 		public unowned string catalog;
+ 		public unowned string def;
+ 		public ulong length;
+ 		public ulong max_length;
+ 		public uint flags;
+ 		public uint decimals;
+ 		public FieldType type;
  	}
  	public unowned string get_client_info ();
  	public ulong get_client_version ();
  	public void debug (string msg);
